### PR TITLE
[ELY-1833] Add ability to KeyStoreCredentialStore to use Provider[] for passwords.

### PR DESCRIPTION
**Pull request by ivassile**
https://issues.jboss.org/browse/ELY-1833
https://issues.jboss.org/browse/EAP7-1360

### CODE REVIEW

File: elytron/ELY-1833_KeyStoreCredentialStore_Provider[]_for_passwords.adoc [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+=== Non-Requirements
+
+== Test Plan
+
+== Community Documentation
+
+New option added to Elytron Tool - Credential Store command:
+Long option (no arguments): --providers-for-passwords
+Short option (no arguments): -w
+Description: Use the specified Provider(s) algorithm implementation for passwords.
```
Comment (OndrejKotek) [inline elytron/ELY-1833_KeyStoreCredentialStore_Provider[]_for_passwords.adoc:64]: What are the allowed values or how to determine those?

File: elytron/ELY-1833_KeyStoreCredentialStore_Provider[]_for_passwords.adoc [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+* mailto:{email}[{author}]
+* mailto:darran.lofthouse@redhat.com[Darran Lofthouse]
+* mailto:fjuma@redhat.com[Farah Juma]
+
+=== QE Contacts
+
+=== Testing By
+
+[ ] Engineering
```
Comment (OndrejKotek) [inline elytron/ELY-1833_KeyStoreCredentialStore_Provider[]_for_passwords.adoc:36]: Engineering tested please

File: elytron/ELY-1833_KeyStoreCredentialStore_Provider[]_for_passwords.adoc (hunk 1 above)
Comment (ivassile) [reply elytron/ELY-1833_KeyStoreCredentialStore_Provider[]_for_passwords.adoc:72]: The option is just a flag without a value.

File: elytron/ELY-1833_KeyStoreCredentialStore_Provider[]_for_passwords.adoc [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+=== Affected Projects or Components
+
+ * wildfly-elytron
+ * Elytron Tool
+ 
+=== Other Interested Projects
+
+== Requirements
+Modify KeyStoreCredentialStore implementation to use the Provider[] passed in during initialisation for PasswordFactory interaction.
+Add new option to Credential Store command to utilize the new feature.
```
[the remark this answers is not in the document]
Comment (ivassile) [reply elytron/ELY-1833_KeyStoreCredentialStore_Provider[]_for_passwords.adoc]: Added description from the Overview section. You can find a use case in https://issues.redhat.com/browse/ELY-1976

File: elytron/ELY-1833_KeyStoreCredentialStore_Provider[]_for_passwords.adoc [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+ * Elytron Tool
+ 
+=== Other Interested Projects
+
+== Requirements
+* The KeyStoreCredentialStore implementation does not use the Provider[] passed in during initialisation for PasswordFactory interaction. However it does use it for the underlying KeyStore. Modify KeyStoreCredentialStore implementation to use the Provider[] passed in during initialisation for PasswordFactory interaction.
+* Add new option to Credential Store command to utilize the new feature.
+Long option (no arguments): --providers-for-passwords
+Short option (no arguments): -w
+Description: Use the specified Provider(s) algorithm implementation for passwords.
```
Comment (OndrejKotek) [inline elytron/ELY-1833_KeyStoreCredentialStore_Provider[]_for_passwords.adoc:52]: Do I understand it well that it's a boolean option? When present it's true by default? And when not present the old behavior takes place? (Which addresses the backward compatibility...)

Comment (OndrejKotek) [reply elytron/ELY-1833_KeyStoreCredentialStore_Provider[]_for_passwords.adoc:52]: I have realized that you provided the explanation in one of the resolved threads. Can you add it also to the analysis please?

**Comment by ivassile** [comment]
Closing since similar functionality was already added in ELY-1976

**Pull request closed without merging**